sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,654 @@
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param index Pad index.
+ * @param type Pad type (MIO or DIO).
+ * @param mode Pad deep sleep mode.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_toggle_result_t dif_pinmux_pad_sleep_enable(
+    dif_pinmux_t *pinmux, dif_pinmux_index_t index, dif_pinmux_pad_kind_t type,
+    dif_pinmux_sleep_mode_t mode);
+
+/**
+ * Disables deep sleep mode of a particular pad.
+ *
+ * NOTE:
+ * After the chip has come back from deep sleep, the mode of a pad persists
+ * until is explicitly cleared by `dif_pinmux_pad_sleep_clear_state`. Calling
+ * `dif_pinmux_pad_sleep_enable` will configure a pad mode for the next deep
+ * sleep, however it will not change the current mode.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param index Pad index.
+ * @param type Pad type (MIO or DIO).
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_toggle_result_t dif_pinmux_pad_sleep_disable(
+    dif_pinmux_t *pinmux, dif_pinmux_index_t index, dif_pinmux_pad_kind_t type);
+
+/**
+ * Returns the state of a particular pad.
+ *
+ * When the pad is in deep sleep mode, the mode persists until is explicitly
+ * cleared via `dif_pinmux_pad_sleep_clear_state`.
+ *
+ * @param padmux A Pin Multiplexer handle.
+ * @param index Pad index.
+ * @param type Padring pad type (MIO or DIO).
+ * @param[out] in_sleep_mode Pad state, `true` when in deep sleep mode.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_result_t dif_pinmux_pad_sleep_get_state(const dif_pinmux_t *pinmux,
+                                                   dif_pinmux_index_t index,
+                                                   dif_pinmux_pad_kind_t type,
+                                                   bool *in_sleep_mode);
+
+/**
+ * Clears deep sleep mode for a particular pad.
+ *
+ * When deep sleep mode is enabled for a pad, and the device has entered deep
+ * sleep mode; upon wake-up, the deep sleep status of this pad can only be
+ * cleared by calling this function. Re-configuring this pad will only take
+ * effect during the next deep sleep.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param index Pad index.
+ * @param type Padring pad type (MIO or DIO).
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_result_t dif_pinmux_pad_sleep_clear_state(
+    const dif_pinmux_t *pinmux, dif_pinmux_index_t index,
+    dif_pinmux_pad_kind_t type);
+
+/**
+ * Enables edge triggered wake-up mode detection for a particular detector.
+ *
+ * @param pinmux A Pin Multiplexer handle.
+ * @param detector_index A detector index.
+ * @param config A wake-up detector configuration.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_pinmux_toggle_result_t dif_pinmux_wakeup_detector_enable_edge(
+    const dif_pinmux_t *pinmux, dif_pinmux_index_t detector_index,
```

Comment:
nit: You could drop `_index` from `detector_index`. The type name makes it clear that its an index. Up to you.

Reply:
Done.